cstrml/swork/src/lib.rs

```diff
@@ -477,30 +541,28 @@ impl<T: Trait> Module<T> {
             return;
         }
 
-        let mut total_used = 0;
-        let mut total_free = 0;
+        let mut total_used = 0u128;
+        let mut total_free = 0u128;
 
         log!(
             trace,
             "🔒 Loop all identities and update the workload map for slot {:?}",
             reported_rs
         );
         // 2. Loop all identities and get the workload map
-        // TODO: avoid iterate all identities
-        let workload_map: Vec<(T::AccountId, u128)> = <IdBonds<T>>::iter().map(|(reporter, ids)| {
-            let mut workload = 0;
-            for id in ids {
-                // 2.1 Calculate reporter's own reserved and used space
-                let (free, used) = Self::get_workload(&reporter, &id, current_rs);
-
-                // 2.2 Update total
-                total_used = total_used.saturating_add(used);
-                total_free = total_free.saturating_add(free);
-                workload = workload.saturating_add(used).saturating_add(free);
+        let mut workload_map= BTreeMap::new();
+        for (reporter, id) in <Identities<T>>::iter() {
+            let (free, used) = Self::get_workload(&reporter, &id.anchor, current_rs);
+            total_used = total_used.saturating_add(used);
+            total_free = total_free.saturating_add(free);
+            let mut owner = reporter;
+            if let Some(group) = id.group {
+                owner = group;
             }
+            let workload = workload_map.get(&owner).unwrap_or(&0u128).saturating_add(used).saturating_add(free);
+            workload_map.insert(owner, workload);
```

Comment:
Add `TODO` in here, since we may need to pass `used` and `free` seperately in the future.